cinema/permissions.py

```diff
@@ -0,0 +1,15 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    """
+    The request is authenticated as an admin
+    - read/write, if as an user - read-only request.
+    """
+
+    def has_permission(self, request, view):
+        return bool(
+            request.method in SAFE_METHODS
+            and request.user
+            and request.user.is_authenticated
+        ) or (request.user and request.user.is_staff)
```

Comment:
The logic in the `has_permission` method seems to be incorrect. According to the class docstring, if the user is authenticated as an admin, they should have read/write access, and if they are a regular authenticated user, they should have read-only access. However, the current logic allows read-only access to any authenticated user and read/write access only to staff. You might want to adjust the conditions to reflect the intended behavior.